policy-engine/src/openapi.rs

```diff
@@ -0,0 +1,65 @@
+use actix_web::{HttpRequest, HttpResponse};
+use openapiv3::OpenAPI;
+use AppState;
+
+const SPEC: &str = include_str!("openapiv3.json");
+
+pub(crate) fn index(req: HttpRequest<AppState>) -> HttpResponse {
+    let path_prefix = &req.state().path_prefix;
+
+    let mut spec_object: OpenAPI = match serde_json::from_str(SPEC) {
+        Ok(o) => o,
+        Err(e) => {
+            let e = format_err!("Could not deserialize to OpenAPI object: {}", e);
+            error!("{}", e);
+            return HttpResponse::from_error(e.into());
+        }
+    };
+
+    // Prefix all paths with `path_prefix`
+    spec_object.paths = rewrite_paths(&spec_object.paths, path_prefix);
+
+    match serde_json::to_string(&spec_object) {
+        Ok(s) => HttpResponse::from(s),
+        Err(e) => {
+            let e = format_err!("Could not serialize OpenAPI object: {}", e);
+            error!("{}", e);
+            HttpResponse::from_error(e.into())
+        }
+    }
+}
+
+fn rewrite_paths(paths: &openapiv3::Paths, path_prefix: &str) -> openapiv3::Paths {
```

Comment:
This allocate a whole new `Paths`. Is it possible to re-arrange this to take ownership of it and consume the iterator into the result? Or alternatively, prefix the paths in place?

Reply:
I wish we could rely on some CoW mechanism, or maybe we can? I'll look into your suggestion and report back.

Reply:
Current moving in and out seems clean and nice enough to me, thanks!